### PR TITLE
Ability to add custom job held text

### PR DESCRIPTION
**Pull request by haxzie-xx**
In reference to #4 to introduce the ability to add custom job held text

- [X]  Update layout to add a custom text field
- [X] Create methods to handle the custom texts
- [X] Replace direct call to `Messages` to `WorkingHoursConfig` for the custom text.

### CODE REVIEW

**Comment by haxzie** [comment]
@jeffpearce I have some doubts on how to pass the user provided String to `CauseOfBlockage.fromMessage()` since it takes a `Localizable`. Should I replace the call to `Messages` with a direct call to getters in `WorkingHoursConfig`?

**Comment by jeffpearce** [comment]
I think that could work. Could the getter return the string from the resources if it's not set in the UI?

**Comment by haxzie** [comment]
Cool, I'd be able to do that. I am kinda stuck at passing the custom String to `CauseOfBlockage.fromMessage()` because the method takes a `Localizable`. Is there any way I can replace the strings loaded into `Messages.java` dynamically with the user provided custom String?
https://github.com/jenkinsci/working-hours-plugin/blob/69ef72cd1a0a33c998f8fc1e9b93323449f21fd7/src/main/java/org/jenkinsci/plugins/workinghours/WorkingHoursQueueTaskDispatcher.java#L84

**Comment by jeffpearce** [comment]
Hi @haxzie . Sorry for the delay. I searched JenkinsCI, and I found at least one example that overrides CauseOfBlockage to return a string rather then a formatted message. Do you think something like that could work?

**Comment by haxzie** [comment]
Yes, that'll help.

**Comment by haxzie** [comment]
Sorry for the delay @jeffpearce , it was rather an easy fix! 
this closes #2

File: src/main/java/org/jenkinsci/plugins/workinghours/WorkingHoursConfig.java [hunk 1]
```diff
@@ -131,4 +136,24 @@ public void setExcludedDates(
         this.excludedDates = value;
         save();
     }
+
+    /**
+     * Gets the message to be displayed when the job is blocked
+     *
+     * @return default message if the value is null
```
Comment (jeffpearce) [inline src/main/java/org/jenkinsci/plugins/workinghours/WorkingHoursConfig.java]: nit: it really returns the custom message, or the default message if the custom message is null

File: src/main/java/org/jenkinsci/plugins/workinghours/WorkingHoursConfig.java [hunk 2]
```diff
@@ -131,4 +136,24 @@ public void setExcludedDates(
...
+     */
+    public String getCustomJobHoldText() {
```
Comment (jeffpearce) [inline src/main/java/org/jenkinsci/plugins/workinghours/WorkingHoursConfig.java]: I think we should rename this to make the code easier to read for people not familiar - when I read this it sounds like it only handles the custom text, but that's only 1/2 of what it does. 

I actually think `getJobHoldText` would be clearer, but I'm open to suggestions

Comment (haxzie) [reply src/main/java/org/jenkinsci/plugins/workinghours/WorkingHoursConfig.java]: Yeah, `getJobHeldText` sounds much better :+1:

**Comment by haxzie** [comment]
Changed the method name and comments in https://github.com/jenkinsci/working-hours-plugin/pull/4/commits/ea5cc0f460db076ec65c3f711f4968238967645f :+1: 
from `getCustomJobHeldText()` to `getJobHoldText()`

**Comment by jeffpearce** [comment]
HI @haxzie. Thanks for implementing this! I'm going to do a release soon, which will including it.

**Pull request merged**